org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -58,4 +61,15 @@ public static String getJavaSourceLevel(IProject project) {
 		return options.get(JavaCore.COMPILER_SOURCE);
 	}
 
+	public static List<IProject> getGradleProjects() {
+		IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
```

Comment:
Stream.of(projects).filter(ProjectUtils ::isGradleProject).collect(Collectors.toList())